hail/python/hailtop/hailctl/config/utils.py

```diff
@@ -0,0 +1,149 @@
+                  f'to assign you the StorageAdmin role in Google Cloud Storage or ask them to update the bucket {bucket} on your behalf.'
+            if lifecycle_days:
+                msg += f'Update the bucket to have a lifecycle policy of {lifecycle_days} days.'
+            if labels_str:
+                msg += f'Update the bucket to have labels: {labels_str}'
+            raise InsufficientPermissions(msg) from e
+        raise
+
+
+async def grant_service_account_bucket_read_access(project: Optional[str], bucket: str, service_account: str, verbose: bool):
+    from hailtop.utils import CalledProcessError, check_shell  # pylint: disable=import-outside-toplevel
+    if project:
+        project = f'--project {project}'
+    else:
+        project = ''
+    try:
+        service_account_member = f'serviceAccount:{service_account}'
+        await check_shell(f'gcloud {project} storage buckets add-iam-policy-binding gs://{bucket} --member {service_account_member} --role "roles/storage.objectViewer"',
+                          echo=verbose)
+    except CalledProcessError as e:
+        if 'does not have storage.buckets.getIamPolicy access to the Google Cloud Storage bucket' in e.stderr.decode('utf-8'):
+            msg = f'ERROR: You do not have the necessary permissions to set permissions for bucket {bucket} in project {project}. Ask a project administrator ' \
+                  f'to assign you the StorageIAMAdmin role in Google Cloud Storage or ask them to update the bucket {bucket} on your behalf by giving ' \
+                  f'service account {service_account} the role "roles/storage.objectViewer".'
+            raise InsufficientPermissions(msg) from e
+        raise
+
+
+async def grant_service_account_bucket_write_access(project: Optional[str], bucket: str, service_account: str, verbose: bool):
+    from hailtop.utils import CalledProcessError, check_shell  # pylint: disable=import-outside-toplevel
+    if project:
+        project = f'--project {project}'
+    else:
+        project = ''
+    try:
+        service_account_member = f'serviceAccount:{service_account}'
+        await check_shell(f'gcloud {project} storage buckets add-iam-policy-binding gs://{bucket} --member {service_account_member} --role "roles/storage.objectCreator"',
+                          echo=verbose)
+    except CalledProcessError as e:
+        if 'does not have storage.buckets.getIamPolicy access to the Google Cloud Storage bucket' in e.stderr.decode('utf-8'):
+            msg = f'ERROR: You do not have the necessary permissions to set permissions for bucket {bucket} in project {project}. Ask a project administrator ' \
+                  f'to assign you the StorageIAMAdmin role in Google Cloud Storage or ask them to update the bucket {bucket} on your behalf by giving ' \
+                  f'service account {service_account} the role "roles/storage.objectCreator".'
+            raise InsufficientPermissions(msg) from e
+        raise
```

Comment:
We can replace these two functions with one function that also accepts a `role`.